DIPs/DIPNNNN.md

```diff
@@ -0,0 +1,359 @@
+   make it possible to adjust semantics or even completely revert the feature
+   based on practical experience - without any risk that it will affect other
+   users of the feature.
+
+As an extra benefit, such a facility can provide a deprecation path for
+introducing new language keywords if the need arises, though this use case is
+beyond the scope of this DIP and won't be evaluated in detail.
+
+### Comparison against other languages
+
+Existing programming languages can be separated in 3 main groups with regard
+to this issue:
+
+#### With single global namespace
+
+Examples: C, early PHP
+
+Such languages tend to rely on [using obscure names](#using-more-obscure-name)
+that makes symbol clash less likely but in general the problem is still present.
+When the problem manifests, situation becomes much worse than in D, because such
+languages lack tools to disambiguate name clash explicitly.
+
+#### With namespaces, allowing non-qualified access
+
+Examples: C++, C#
+
+This category is most similar to D, meaning that the language provides tools to
+disambiguate any names but at the same time allows imports with non-qualified
+access to imported symbols.
+
+In D this is default `import` behavior.
+
+In C++ : `using namespace Namespace.Subnamespace`.
+
+In C# : `using Namespace.Subnamespace`
+
+The importance of problem varies depending on the prevalent idioms used in the
+language and other available tools. For example, C# documentation
+[recommends](https://msdn.microsoft.com/en-us/library/dfb3cx8s.aspx) using
+dedicated namespace access operator `::` instead of uniform `.` to reduce the
+risk of clashes.
+
+However, there are a few aspects specific to D that make issues somewhat more
+important:
+
+- In all mentioned languages but D, a common convention is to
+  only use unqualified access for symbols in standard library. In D plain
+  `import` is the most widespread way to use all libraries, including the dub
+  registry.
+- The D runtime features the module `object.d` which is implicitly imported
+  everywhere, greatly increasing chance for name clashes for any symbol in it.
+
+#### With namespaces, restricted non-qualified access
+
+Examples: Rust, Go
+
+In both Rust and Go normal imports still require using module name for
+qualification. For example, in Rust `use Namespace::Subnamespace::Module` still
+requires to explicitly write `Module::foo()` to call `foo` function.
+
+Both languages provide a way to enable unqualified access (`import . "Module"`
+in Go, `use Namespace::Module::*` in Rust) but this feature is heavily
+discouraged from usage in production projects exactly because of the problems
+described in this DIP.
+
+Using qualified lookuos consistently solves the issue described in this DIP but
+such an approach is too different to established idiomatic D coding style to be
+considered.
+
+### Breaking changes / deprecation process
+
+The proposed semantics are purely additive and do not affect existing code.
```

Comment:
Well, this is considering `@future` is not yet "exported" to user code, otherwise it will break code from users already defining the UDA `@future`, right?